src/Location.ml

```diff
@@ -0,0 +1,62 @@
+open Base
+open Lexing
+
+type t =
+  | NoSource
+  | Source of {
+      filename : string; [@equal.ignore]
+      start_line : int; [@equal.ignore]
+      start_column : int; [@equal.ignore]
+      end_line : int; [@equal.ignore]
+      end_column : int; [@equal.ignore]
+    }
+[@@deriving equal, sexp]
+
+type 'a located = { data : 'a; loc : t } [@@deriving equal, sexp]
+
+let mk s_pos e_pos =
+  Source
+    {
+      filename = s_pos.pos_fname;
+      start_line = s_pos.pos_lnum;
+      start_column = s_pos.pos_cnum;
+      end_line = e_pos.pos_lnum;
+      end_column = e_pos.pos_cnum;
+    }
+
+let locate data s_pos e_pos =
+  let loc = mk s_pos e_pos in
+  { data; loc }
+
+let mkLocated ?(loc = NoSource) data = { data; loc }
+
+let showPos pos' =
```

Comment:
Please stick to the OCaml naming conventions.

Reply:
rename